DemoApp/CallKitService.swift

```diff
@@ -187,22 +189,24 @@ class CallKitService: NSObject, CXProviderDelegate, @unchecked Sendable {
     
     private func subscribeToCallEvents() {
         Task {
-            for await event in streamVideo.callEvents() {
-                switch event {
-                case .ended(_):
-                    endCurrentCall()
-                case .rejected(let callInfo):
-                    if callInfo.user?.id == streamVideo.user.id {
-                        endCurrentCall()
-                    } else if callInfo.user?.id == createdBy?.id {
-                        endCurrentCall()
-                    }
-                case .accepted(let callInfo):
-                    if callInfo.user?.id == streamVideo.user.id && state == .idle {
+            for await wsEvent in streamVideo.subscribe() {
+                if let event = callEventsHandler.checkForCallEvents(from: wsEvent) {
+                    switch event {
+                    case .ended(_):
                         endCurrentCall()
+                    case .rejected(let callInfo):
+                        if callInfo.user?.id == streamVideo.user.id {
+                            endCurrentCall()
+                        } else if callInfo.user?.id == createdBy?.id {
```

Comment:
Shouldn't we simply ignore the call and not do anything in this case?

Reply:
not sure I understand what you mean? We need to reject a call on all devices, if it's rejected by one.